test grafici

lorenzocorallo opened the pull request:
# NON MERGE

Commit:
Update manifest.json

Commit:
Update CNAME and CNAME

Commit:
Update Table.tsx

angeousta committed:
Merge pull request #5 from PoliNetworkOrg/4-searchbar

fix #4

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update index-31109e7f.js, index-4a824cd6.css, logo3000-8254a9f4.webp, and 5 more files

Commit:
Update .gitignore

Commit:
Update Viewer.tsx

angeousta committed:
Merge pull request #13 from PoliNetworkOrg/12-aggiungere-tasto-o-una-pagina-apposita-per-scaricare-i-dati-in-formati-user-friendly-es-csv

fix #12

lorenzocorallo committed:
feat(ci): add EndBug/pages-preview workflow

- change node version to lts
- support for PR previews

angeousta committed:
Merge pull request #16 from PoliNetworkOrg/add-preview

feat(ci): add EndBug/pages-preview workflow

lorenzocorallo committed:
fix(ci): correctly use PUBLIC_URL basename (#17)

* fix: remove previous added router baseName

* fix(ci): use preview workflow on each branch

* fix: add env.PUBLIC_URL basename to react-router

lorenzocorallo committed:
docs: add preview url and description

angeousta committed:
Merge pull request #18 from PoliNetworkOrg/update-readme

docs: add preview url and description

lorenzocorallo committed:
fix(ci): update preview triggers (#19)

lorenzocorallo committed:
fix(ci): re-add github pages deploy (#20)

lorenzocorallo committed:
fix(ci): add build env.PUBLIC_URL for gh-pages (#21)

lorenzocorallo committed:
fix(ci): update build env.PUBLIC_URL for gh-pages (#22)

lorenzocorallo committed:
style: fix select buttons layout on mobile (#23)

lorenzocorallo committed:
style: add different bg color to table even rows (#24)

lorenzocorallo committed:
style: capitalize words of courses' name

lorenzocorallo committed:
style: add circle button for CSV download

angeousta committed:
Update package.json and pnpm-lock.yaml

angeousta committed:
Update index.html

angeousta committed:
Update About.tsx

angeousta committed:
Update Privacy.tsx

angeousta committed:
Update index.html and About.tsx

lorenzocorallo committed:
fix: refactor convert-to-csv function

lorenzocorallo committed:
Merge pull request #29 from PoliNetworkOrg/move-csv-functions

fix: refactor convert-to-csv function

lorenzocorallo committed:
Merge branch 'main' into update-con-info

lorenzocorallo committed:
Merge pull request #27 from PoliNetworkOrg/update-con-info

Update con info

angeousta committed:
Update Viewer.tsx

angeousta committed:
Merge pull request #31 from PoliNetworkOrg/angeousta-patch-1

Update Viewer.tsx

angeousta committed:
Merge branch 'main' into 12-csv-btn

angeousta committed:
Merge pull request #26 from PoliNetworkOrg/12-csv-btn

feat: CSV download

angeousta committed:
Update package.json and pnpm-lock.yaml

angeousta committed:
Merge branch 'main' into 9-texts

angeousta committed:
Merge pull request #25 from PoliNetworkOrg/9-texts

style: better courses buttons

angeousta committed:
Update preview.yml

angeousta committed:
Merge pull request #35 from PoliNetworkOrg/9-texts

Update preview.yml

lorenzocorallo committed:
feat: add new json types

lorenzocorallo committed:
feat: add Path class with join method

lorenzocorallo committed:
deps: add axios for fetching

lorenzocorallo committed:
chore: rename data.ts to store.ts

angeousta committed:
Update README.md

angeousta committed:
Merge pull request #39 from PoliNetworkOrg/angeousta-patch-1

Update README.md

angeousta committed:
Update About.tsx

lorenzocorallo committed:
fix: update data types

lorenzocorallo committed:
feat: add Alert component

lorenzocorallo committed:
data: add github csharp output base url

lorenzocorallo committed:
feat: start writing Data class and create its context

lorenzocorallo committed:
fix: update Store import in Viewer

lorenzocorallo committed:
feat: add global settings for react-icons

angeousta committed:
Update About.tsx and Privacy.tsx

lorenzocorallo committed:
feat: add loading state in App

lorenzocorallo committed:
fix: change School type construction

angeousta committed:
Update About.tsx

lorenzocorallo committed:
style: update Alert

angeousta committed:
Merge pull request #41 from PoliNetworkOrg/about

About

lorenzocorallo committed:
chore: move Alert Level type definition

lorenzocorallo committed:
fix: Data infinite fetching

lorenzocorallo committed:
debug: axios url

lorenzocorallo committed:
deps: remove axios

lorenzocorallo committed:
fix: correct url parsing

lorenzocorallo committed:
fix: correct icons in Alert

lorenzocorallo committed:
feat: add Test route to view components

lorenzocorallo committed:
feat: add View component

lorenzocorallo committed:
debug: remove data url print

angeousta committed:
Update 404.html

angeousta committed:
Update 404.html and 404.html

angeousta committed:
Update gh-pages.yml, index.html, manifest.json, and 3 more files

lorenzocorallo committed:
feat: implement basic csharp

lorenzocorallo committed:
fix: remove errors

lorenzocorallo committed:
feat: add navigation breadcrumb

lorenzocorallo committed:
try fix gh pages

lorenzocorallo committed:
try fix gh pages 2

lorenzocorallo committed:
browser router

lorenzocorallo committed:
hash router

lorenzocorallo committed:
Merge pull request #51 from PoliNetworkOrg/hash-router

Hash router

lorenzocorallo committed:
Merge branch 'main' into move-to-csharp

angeousta committed:
Update Homepage.tsx

angeousta committed:
Update Homepage.tsx

lorenzocorallo committed:
style: update View selects

lorenzocorallo committed:
style: warning low

lorenzocorallo committed:
Merge branch 'move-to-csharp' into warning

angeousta committed:
Merge pull request #52 from PoliNetworkOrg/warning

Warning

angeousta committed:
Merge branch 'move-to-csharp' into remove-beta

lorenzocorallo committed:
style: homepage buttons in grid 2-cols

lorenzocorallo committed:
fix: 404 in public

angeousta committed:
Merge branch 'move-to-csharp' into router-404

angeousta committed:
Merge branch 'move-to-csharp' into remove-beta

lorenzocorallo committed:
style: breadcrumb mobile

lorenzocorallo committed:
fix: phase breadcrumb

lorenzocorallo committed:
fix: move matricola to last

lorenzocorallo committed:
feat: add csv download

lorenzocorallo committed:
style: change to grid 2-cols buttons

angeousta committed:
Merge branch 'move-to-csharp' into router-404

lorenzocorallo committed:
fix: build

lorenzocorallo committed:
style: add little mobile to table view

angeousta committed:
Merge pull request #43 from PoliNetworkOrg/move-to-csharp

Move to csharp

lorenzocorallo committed:
fix: add location to course

lorenzocorallo committed:
Merge pull request #55 from PoliNetworkOrg/fix-naems

fix: add location to course

lorenzocorallo committed:
fix: fake filename redirects to home or prev choose

lorenzocorallo committed:
Merge pull request #57 from PoliNetworkOrg/fix-fake-filename

fix: fake filename redirects to home or prev choose

lorenzocorallo committed:
fix: remove key errors

lorenzocorallo committed:
Merge pull request #58 from PoliNetworkOrg/fix-key-errors

fix: remove key errors

lorenzocorallo committed:
feat: add 404 for internal router

lorenzocorallo committed:
Merge pull request #59 from PoliNetworkOrg/react-router-404

feat: add 404 for internal router

lorenzocorallo committed:
Merge branch 'main' into remove-beta

lorenzocorallo committed:
Merge pull request #47 from PoliNetworkOrg/remove-beta

remove beta

lorenzocorallo committed:
Create CNAME

lorenzocorallo committed:
Merge pull request #60 from PoliNetworkOrg/cname

Create CNAME

lorenzocorallo committed:
Update CNAME

lorenzocorallo committed:
Merge pull request #61 from PoliNetworkOrg/fix-cname

Update CNAME

lorenzocorallo committed:
Update gh-pages.yml

lorenzocorallo committed:
Merge pull request #62 from PoliNetworkOrg/fix-deploy

Update gh-pages.yml

lorenzocorallo committed:
Merge branch 'main' into router-404

lorenzocorallo committed:
Merge pull request #44 from PoliNetworkOrg/router-404

Update 404.html

angeousta committed:
Update README.md

angeousta committed:
Merge pull request #63 from PoliNetworkOrg/angeousta-patch-1

Update README.md

lorenzocorallo committed:
Update 404.html

lorenzocorallo committed:
Merge pull request #64 from PoliNetworkOrg/fix-404

Update 404.html

lorenzocorallo committed:
fix: check for empty table in csv

lorenzocorallo committed:
fix: empty course title or location

lorenzocorallo committed:
fix: better fix for empty course location

lorenzocorallo committed:
fix: view empty table

lorenzocorallo committed:
Merge branch 'main' into fix

lorenzocorallo committed:
Merge pull request #65 from PoliNetworkOrg/fix

fix: multi errors

angeousta committed:
Update README.md

angeousta committed:
Merge pull request #67 from PoliNetworkOrg/angeousta-patch-2

Update README.md

angeousta committed:
Update README.md

angeousta committed:
Merge pull request #68 from PoliNetworkOrg/angeousta-patch-3

Update README.md

lorenzocorallo committed:
clean old stuff

lorenzocorallo committed:
Merge pull request #70 from PoliNetworkOrg/cleanup

clean old stuff

lorenzocorallo committed:
release: v1.0.0 (#71)

* release: v1.0.0

* ci: deploy gh-pages only on release

* docs: update README

lorenzocorallo committed:
remove old file

lorenzocorallo committed:
Merge pull request #72 from PoliNetworkOrg/release-v1

remove old file

lorenzocorallo committed:
feat: cache ranking once downloaded

lorenzocorallo committed:
fix: use ranking spaced-name in breadcrumb

lorenzocorallo committed:
feat: add phase list getter from ranking and course

lorenzocorallo committed:
style: add custom scrollbar to breadcrumb

lorenzocorallo committed:
feat: complete the phase switcher from table view

lorenzocorallo committed:
chore: move Viewer index component to index file

lorenzocorallo committed:
fix: better url check and 404 redirects

lorenzocorallo committed:
fix: use original ranking phase in breadcrumb

lorenzocorallo committed:
refactor: separate functions for caching rankings

lorenzocorallo committed:
Merge pull request #74 from PoliNetworkOrg/69-breadcrumb-name

fix: use ranking spaced-name in breadcrumb

lorenzocorallo committed:
Merge branch 'main' into cache-ranking

lorenzocorallo committed:
Merge pull request #73 from PoliNetworkOrg/cache-ranking

feat: cache ranking once downloaded

lorenzocorallo committed:
Merge branch 'main' into phase-switch-from-course

lorenzocorallo committed:
Merge pull request #75 from PoliNetworkOrg/phase-switch-from-course

feat: add phase switch from course table

lorenzocorallo committed:
refactor: split getRanking function

lorenzocorallo committed:
Merge branch 'main' into cache-ranking

lorenzocorallo committed:
Merge pull request #79 from PoliNetworkOrg/cache-ranking

refactor: split getRanking function

lorenzocorallo committed:
chore: custom labels

lorenzocorallo committed:
Merge pull request #81 from PoliNetworkOrg/custom-labels

chore: custom labels

lorenzocorallo committed:
feat(ci): add sync-labels.yml workflow

lorenzocorallo committed:
Merge pull request #82 from PoliNetworkOrg/ci-labels

feat(ci): add sync-labels.yml workflow

lorenzocorallo committed:
feat: add recharts

lorenzocorallo committed:
feat: charts n.3a + istogramma a caso

angeousta committed:
Merge branch 'main' into grafici

angeousta commented:
merge non da fare per ora

bob27aggiustatutto commented:
![Screenshot_20230706-092033_Brave](https://github.com/PoliNetworkOrg/Rankings/assets/120955097/cda8792c-0cb1-434f-9404-cd713601a793)

Mi piacerebbe che tutta questa parte di codice di ottenimento dei dati stesse su un file sorgente a parte

angeousta committed:
Update MinScorePhases.tsx, VotoCandidatiChart.tsx, and index.tsx

bob27aggiustatutto committed:
Merge pull request #85 from PoliNetworkOrg/grafici-sub-1

Refactor grafici

lorenzocorallo committed:
refactor: use Map instead of Object (Dictionary)

- create a Map class extension CustomMap, including some utils methods
- split data types in json (data strcture coming from json files) and
  parsed (data structure involving map)
- create jsonParser class to parse json (creating our parsed data)
- fix: there was a bug that didn't show some headers in some rankings
  tables; using a new header recognition method available as the "Has"
  type and relative method in the Table component
- make Data class the default export of the file data.ts

lorenzocorallo committed:
chore: rename Index type files including Index word

angeousta committed:
Merge pull request #86 from PoliNetworkOrg/refactor

refactor: use Map instead of Object (Dictionary)

lorenzocorallo committed:
Merge branch 'main' into grafici + stats fixes

lorenzocorallo committed:
feat: add stats year fetching and caching

lorenzocorallo committed:
fix: no graph for empty tables

lorenzocorallo committed:
partial colors fix

lorenzocorallo committed:
init shadcn ui

lorenzocorallo committed:
ui: add button

lorenzocorallo committed:
fix: Timeout in useFilter hooks

lorenzocorallo committed:
revert index.css to main state

lorenzocorallo committed:
feat: use shadcn ui components (Button, Tab, Combobox)

lorenzocorallo committed:
install @tanstack/react-table

lorenzocorallo committed:
feat: use @tanstack/table with shadcn-ui

lorenzocorallo committed:
fix: toFixed method on null value

lorenzocorallo committed:
style: better headers and cell data visualization

angeousta committed:
feat: migrate from `react-router` to `@tanstack/router` (#87)

* install @tanstack/router@beta

* refactor: extract Router and Layout components from App.tsx

* install @tanstack/router-devtools

* feat: migrate from react-router to @tanstack/router

* rename folder pages to routes and create subfolders for each route

* feat: move route def and component to separate files (excluded Viewer)

* remove react-router-dom

* update router and router-devtools pkgs

* fix: imports

* update router pkgs

* update router pkgs

* fix: deep route matching

* fix: remove old import

---------

Co-authored-by: Lorenzo <lorenzo.corallo@outlook.it>

lorenzocorallo committed:
Merge branch 'main' into shadcn-ui

lorenzocorallo committed:
fix: show router devtools only on dev

lorenzocorallo committed:
style: adjust border colors and fix button variant

lorenzocorallo committed:
refactor: move Viewer component to routes

lorenzocorallo committed:
feat: use router loaders to get data

lorenzocorallo committed:
fix: loaders not a function bug

angeousta committed:
Merge pull request #88 from PoliNetworkOrg/shadcn-ui

feat: add @tanstack/table and shadcn-ui

lorenzocorallo committed:
style: make tabs fill space on large screens

angeousta committed:
Merge pull request #91 from PoliNetworkOrg/89-tabs-style

style: make tabs fill space on large screens

lorenzocorallo committed:
fix: course select combobox empty on "Tutti i corsi"

lorenzocorallo committed:
Merge branch 'main' into 90-emppty-couse-select

angeousta committed:
Merge pull request #92 from PoliNetworkOrg/90-emppty-couse-select

fix: course select combobox empty on "Tutti i corsi"

lorenzocorallo committed:
release: v1.1.0 (#93)

lorenzocorallo committed:
fix(ci): gh-pages workflow not deploying (#94)

lorenzocorallo committed:
release: v1.1.1 (#95)

lorenzocorallo committed:
style: add anchor tag base style

lorenzocorallo committed:
style: transform about page style to tailwindcss

lorenzocorallo committed:
style: transform privacy page style to tailwindcss

lorenzocorallo committed:
style: transform homepage style to tailwindcss

lorenzocorallo committed:
style: remove phase from nav breadcrumb

angeousta committed:
Merge pull request #96 from PoliNetworkOrg/refactor-style

style: anchor tag, style-props to tailwindcss, remove phase from breadcrumb

angeousta commented:
![immagine](https://github.com/PoliNetworkOrg/Rankings/assets/132761637/d48f3077-9e17-4e82-a582-c2162dc3fa19)

@lorenzocorallo  se hai tempo fixa i conflitti di questa pr thanks

lorenzocorallo committed:
fix: correct name of phase switch (#106)

lorenzocorallo committed:
fix: Table no data message (#108)

lorenzocorallo committed:
release: v1.1.2 (#110)

lorenzocorallo committed:
style: buttons list on mobile and years shown from most recent (#107)

* style: on chooseYear show the latest year in first position

* style: buttons in list instead of grid on mobile

lorenzocorallo committed:
style: combobox on mobile (#112)

* chore: remove fake componet

* style: fix combobox on mobile

lorenzocorallo committed:
style: better header

lorenzocorallo committed:
fix: title anchor area sized to text only

angeousta committed:
Merge pull request #113 from PoliNetworkOrg/style-header-size

style: better header (sizes, button, anchor areas)

lorenzocorallo committed:
style: better logo ux (#119)

angeousta committed:
Update index.html

angeousta committed:
Merge pull request #121 from PoliNetworkOrg/angeousta-patch-1

seo

lorenzocorallo committed:
style: increase (combobox) popover width (#120)

lorenzocorallo committed:
style: add scroll-area and apply styled scrollbar (#125)

* style: add shadcn/ui scroll-area component

* style: apply styled scrollbar and scroll areas

lorenzocorallo committed:
release: v1.1.3 (#126)

lorenzocorallo committed:
feat: add table filter by "matricola" (#109)

* feat: add shacdn/ui input

* feat: add table filter by "matricola"

lorenzocorallo committed:
release: v1.2.0 (#128)

lorenzocorallo committed:
style(csv-btn): fix on mobile and on table without input filter (#131)

lorenzocorallo committed:
style(Table): even rows with different bg colors (#133)

* style(Table): even rows with different bg colors

* style: adjust light colors

lorenzocorallo committed:
Merge branch 'main' into grafici

lorenzocorallo committed:
feat: save dark mode preference (in localstorage) (#139)

lorenzocorallo committed:
restyle: homepage hierarchy, buttons grid, icons, routes path, header (#136)

* style: redefine homepage hierarchy

* restyle: homepage hierarchy, buttons grid, icons, header, routes path

* style(ButtonGrid): use n cols layout when n < 4

lorenzocorallo committed:
release: v1.3.0 (#141)

lorenzocorallo committed:
feat(CourseSummary): add howManyCanEnroll and howManyStudents props (#144)

lorenzocorallo committed:
style(Footer): add PoliNetwork link, disclaimer and restyle (#138)

* style(Footer): add PoliNetwork link and re-order links

* style: fix order and change PoliNetwork link title

* style: new footer

* style: reduce footer font-size

* style: update footer

lorenzocorallo committed:
feat: add index.html meta seo and theme (#146)

* feat: add index.html meta seo and theme

* fix: change twitter profile reference

* fix: move bg and text color to body tag (ios safari bar)

lorenzocorallo committed:
release: v1.4.0 (#147)

lorenzocorallo committed:
style: fix home text alignment and ButtonGrid rows height on mobile (#145)

* style(Homepage): fix text-alignment on mobile

* style: redefine small screens breakpoints

* style(ButtonGrid): change mobile button height and grid cols

lorenzocorallo committed:
chore: change prettier config (#150)

* chore: change prettier config

* chore: run prettier with new config

lorenzocorallo committed:
Merge branch 'main' into grafici

lorenzocorallo committed:
fix: merge

lorenzocorallo committed:
feat: add instructions in chooseYear and choosePhase (#151)

lorenzocorallo committed:
style: change welcome message and line breaks (#152)

lorenzocorallo committed:
release: v1.5.0 (#153)

angeousta committed:
Merge branch 'main' into grafici

lorenzocorallo committed:
refactor: move to Ranking rankingOrder (#157)

lorenzocorallo committed:
release: v1.6.0 (#158)

lorenzocorallo committed:
style(PhaseSelect): show combobox if screen width < 768px (#161)

lorenzocorallo committed:
release: v1.6.1 (#163)

lorenzocorallo committed:
feat: default location is the biggest one (#162)

lorenzocorallo committed:
feat: capitalisation with italian rules (#164)

* feat: capitalisation with italian rules

* fix: add "Extra-UE" case

lorenzocorallo committed:
release: v1.7.0 (#167)

lorenzocorallo committed:
feat: phase rework (#165)

* feat: partial phase reworking

* feat: probably complete the reworking

* fix: correct Dis/Urb order or "ripescaggio"

* fix: show correct phase label in combobox

* fix: show correct selected phase label in combobox

* style: change "Fase" to "Graduatoria" in phase selector

* feat: add loading spinner once selected year (all rankings are loading)

* style: always show buttons in grid with same col-width

lorenzocorallo committed:
feat(Viewer): add independent phase and ranking switcher (#171)

- create independent phase and ranking switcher in Viewer page
- create PhaseGroup and PhaseGroups types
- create the NO_GROUP group, which is used when no group is found while
  parsing ranking order (no order.primary value)
- split phase getter functions into chunks

lorenzocorallo committed:
release: v1.8.0 (#175)

lorenzocorallo committed:
Merge branch 'main' into grafici

lorenzocorallo committed:
chore: extract types from PhaseLink and StatsBySchoolMap (#178)

angeousta committed:
Merge branch 'main' into grafici

angeousta commented:
@lorenzocorallo c'è rotta la build

lorenzocorallo committed:
fix: correct type of stats in jsonParser

angeousta commented:
conflitti @lorenzocorallo 

![immagine](https://github.com/PoliNetworkOrg/Rankings/assets/132761637/3ed20792-a1af-423d-8e38-0dff8e91c574)
